File: fincapp-backend/src/FincApp.API/DTOs/SyncHealthRecordDto.cs
using System;
using System.Text.Json.Serialization;

namespace FincApp.API.DTOs;

public class SyncHealthRecordDto
{
    [JsonPropertyName("id")]
    public Guid? Id { get; set; }

    [JsonPropertyName("animal_id")]
    public Guid AnimalId { get; set; }

    [JsonPropertyName("symptoms_description")]
    public string SymptomsDescription { get; set; } = string.Empty;

    [JsonPropertyName("diagnosis")]
    public string? Diagnosis { get; set; }

    [JsonPropertyName("treatment_administered")]
    public string? TreatmentAdministered { get; set; }

    [JsonPropertyName("recorded_at")]
    public DateTime? RecordedAt { get; set; }
}

File: fincapp-frontend/mobile-app/app/src/main/java/com/irwi/fincapp/network/dto/SyncHealthRecordDto.java
package com.irwi.fincapp.network.dto;

import com.google.gson.annotations.SerializedName;

public class SyncHealthRecordDto {
    public String id;
    @SerializedName("animal_id") public String animalId;
    @SerializedName("symptoms_description") public String symptomsDescription;
    public String diagnosis;
    @SerializedName("treatment_administered") public String treatmentAdministered;
    @SerializedName("recorded_at") public String recordedAt;
}
